packages/core/types/src/product/common.ts

```diff
@@ -899,22 +899,25 @@ export interface FilterableProductVariantProps
    * Search through the title and different code attributes on the variant
    */
   q?: string
+
   /**
    * The IDs to filter product variants by.
    */
-  id?: string | string[]
+  id?: string | string[] | OperatorMap<string | string[]>
+
   /**
    * The SKUs to filter product variants by.
    */
-  sku?: string | string[]
+  sku?: string | string[] | OperatorMap<string | string[]>
   /**
    * Filter the product variants by their associated products' IDs.
    */
-  product_id?: string | string[]
+  product_id?: string | string[] | OperatorMap<string | string[]>
+
   /**
    * Filter product variants by their associated options.
    */
-  options?: Record<string, string>
+  options?: Record<string, any>
```

Comment:
question: Couldn't this be typed with the actual shape of options?

Reply:
It could be. But, I can see that we have been lenient everywhere else as well. 

Thinking out loud. Was the idea that if someone provides a custom implementation for a given module, they can continue to use the same interface and have their filters still work?

Reply:
PS: We also need to change for FilterableProductProps as the same issue would be there as well

Reply:
My thinking in this case specifically was that we know the shape of options (which is fixed), so we might as well type it accordingly. Is there a reason we shouldn't do that?

Reply:
Ah you beat me to it. Think this is nice :)

Reply:
@riqwan Fixed https://github.com/medusajs/medusa/pull/11441/files#diff-d39b7d61536c3c804e9498463649a01fbcf75b11747e25f6df396537e577a42cR925-R929